controllers/statefulset.go

```diff
@@ -1295,29 +1338,41 @@ func (r *SingleClusterReconciler) initializeSTSStorage(
 	rackState RackState,
 ) {
 	// Initialize sts storage
+	var specVolumes []corev1.Volume
 	for i := range st.Spec.Template.Spec.InitContainers {
+		externalMounts, volumesForMount := r.getExternalStorageMounts(&st.Spec.Template.Spec.InitContainers[i], rackState, st.Spec.Template.Spec.Volumes)
+		specVolumes = append(specVolumes, volumesForMount...)
 		if i == 0 {
 			st.Spec.Template.Spec.InitContainers[i].VolumeMounts = getDefaultAerospikeInitContainerVolumeMounts()
 		} else {
 			st.Spec.Template.Spec.InitContainers[i].VolumeMounts = []corev1.VolumeMount{}
 		}
-
+		st.Spec.Template.Spec.InitContainers[i].VolumeMounts = append(st.Spec.Template.Spec.InitContainers[i].VolumeMounts, externalMounts...)
+		externalDevices, volumesForDevice := r.getExternalStorageDevices(&st.Spec.Template.Spec.InitContainers[i], rackState, st.Spec.Template.Spec.Volumes)
+		specVolumes = append(specVolumes, volumesForDevice...)
 		st.Spec.Template.Spec.InitContainers[i].VolumeDevices = []corev1.VolumeDevice{}
+		st.Spec.Template.Spec.InitContainers[i].VolumeDevices = append(st.Spec.Template.Spec.InitContainers[i].VolumeDevices, externalDevices...)
 	}
 
 	for i := range st.Spec.Template.Spec.Containers {
+		externalMounts, volumesForMount := r.getExternalStorageMounts(&st.Spec.Template.Spec.Containers[i], rackState, st.Spec.Template.Spec.Volumes)
+		specVolumes = append(specVolumes, volumesForMount...)
 		if i == 0 {
 			st.Spec.Template.Spec.Containers[i].VolumeMounts = getDefaultAerospikeContainerVolumeMounts()
 		} else {
 			st.Spec.Template.Spec.Containers[i].VolumeMounts = []corev1.VolumeMount{}
 		}
-
+		st.Spec.Template.Spec.Containers[i].VolumeMounts = append(st.Spec.Template.Spec.Containers[i].VolumeMounts, externalMounts...)
+		externalDevices, volumesForDevice := r.getExternalStorageDevices(&st.Spec.Template.Spec.Containers[i], rackState, st.Spec.Template.Spec.Volumes)
+		specVolumes = append(specVolumes, volumesForDevice...)
 		st.Spec.Template.Spec.Containers[i].VolumeDevices = []corev1.VolumeDevice{}
+		st.Spec.Template.Spec.Containers[i].VolumeDevices = append(st.Spec.Template.Spec.Containers[i].VolumeDevices, externalDevices...)
 	}
 
 	st.Spec.Template.Spec.Volumes = getDefaultSTSVolumes(
 		r.aeroCluster, rackState,
 	)
+	st.Spec.Template.Spec.Volumes = append(st.Spec.Template.Spec.Volumes, specVolumes...)
```

Comment:
Here also, append may lead to duplicate entries. the same volume can be mounted in initContainer and regular container.

Reply:
Made changes to handle this